packages/bundler/src/types/Types.ts

```diff
@@ -1,13 +1,28 @@
+import { ethers, BigNumber } from 'ethers'
+
+
 export type Bundlerconfig = {
     bundlerUrl: string,
     epAddress: string
 }
 
+export type UserOpReceipt = {
+  userOpHash: string,
+  entryPoint: string,
+  sender: string,
+  nonce: number,
+  paymaster: string,
+  actualGasCost: BigNumber,
+  actualGasUsed: BigNumber,
+  success: boolean,
+  reason: string,
+  logs:  Array<ethers.providers.Log>, // The logs generated by this UserOperation (not including logs of other UserOperations in the same bundle)
+  receipt: ethers.providers.TransactionReceipt
+}
+
 export type UserOpResponse = {
```

Comment:
Make UserOpResponse a class which has state of userOpHash and a wait method that has the context of userOpHash for which it needs to fetch the UserOpReceipt